src/EditorFeatures/Text/Extensions.SnapshotSourceText.cs

```diff
@@ -334,36 +335,45 @@ public override IReadOnlyList<TextChangeRange> GetChangeRanges(SourceText oldTex
                     }
                 }
 
-                var oldSnapshot = oldText.FindCorrespondingEditorTextSnapshot();
-                var newSnapshot = this.FindCorrespondingEditorTextSnapshot();
+                var oldSnapshot = oldText.TryFindCorrespondingEditorTextImage();
+                var newSnapshot = this.TryFindCorrespondingEditorTextImage();
                 return GetChangeRanges(oldSnapshot, oldText.Length, newSnapshot);
             }
 
-            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextSnapshot oldSnapshot, int oldTextLength, ITextSnapshot newSnapshot)
+            private IReadOnlyList<TextChangeRange> GetChangeRanges(ITextImage oldImage, int oldTextLength, ITextImage newImage)
             {
-                if (oldSnapshot == null ||
-                    newSnapshot == null ||
-                    oldSnapshot.TextBuffer != newSnapshot.TextBuffer)
+                var oldSnapshot = oldImage == null ? null : TextBufferMapper.TryFindEditorSnapshot(oldImage);
+                var newSnapshot = newImage == null ? null : TextBufferMapper.TryFindEditorSnapshot(newImage);
```

Comment:
this might be a problem since closed file will not have editor snapshot saved. so all changes to closed file now become whole file change. before even closed file had buffer but not anymore for images.

@olegtk @dpugh what is best way to see whether 2 ITextImage belong to same Buffer(?)